mantidproject/mantid
# Make workbench progress bars display the time remaining for slow algs

**Description of work.**

Includes:
1. Passing the message down the pretty long chain to the widget
2. Updating Pause to output  these messages
3. Removing the ability to access algorithm progress from the Algorithm Observer in Python.  Not used now and a known performance issue.


**To test:**
1. Start workbench
1. Run Pause with a duration off say 30
1. Look at the progress bar in the algorithmToolbox, and under the details button, see the time remaining in the progress bar text.
1. Try a duration of 90, and 9000.  Feel free to cancel them once you see the values are formatted well.

Fixes #25803


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
